### PR TITLE
Sof 150 flight config params and validation

### PR DESCRIPTION
**Pull request by mmb186**

### CODE REVIEW

File: OnLand/ControlLaptop/static/assets/js/FlightProfile.js [hunk 1]
```diff
@@ -13,11 +13,52 @@ function showSuccessConfigurationUpdate(from, align){
 	});
 }
 
+function validConfigFormValues() {
+  let formHasError = true;
```
Comment (lwaghorn) [inline OnLand/ControlLaptop/static/assets/js/FlightProfile.js]: I dont think this variable name makes sense. formaHasError should start false and then flip when an error is found. But you want to return true for a good form. So maybe let noErrors = True?

File: OnLand/ControlLaptop/static/assets/js/FlightProfile.js [hunk 2]
```diff
@@ -13,11 +13,52 @@ function showSuccessConfigurationUpdate(from, align){
...
+  }
+  if (isNaN($('#retrieval-timout').val())) { // Validated Retrieval timeout
+    formHasError = false;
+    $('#retrieval-timout').parent().addClass('has-error');
+  }
+  if (isNaN($('#max-flight-time').val())) { // Validated max-flight time
+    formHasError = false;
+    $('#max-flight-time').parent().addClass('has-error');
+  }
+  if (isNaN($('#motor-speed').val())) { // Validated motor speed
```
Comment (lwaghorn) [inline OnLand/ControlLaptop/static/assets/js/FlightProfile.js]: motor speed > 0

File: OnLand/ControlLaptop/static/assets/js/FlightProfile.js [hunk 3]
```diff
@@ -13,11 +13,52 @@ function showSuccessConfigurationUpdate(from, align){
...
+  if (!(/^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/.test($('#pod-ip').val())))
+  {
+    formHasError = false;
+    $('#pod-ip').parent().addClass('has-error');
+  }
+  if (isNaN($('#retrieval-timout').val())) { // Validated Retrieval timeout
+    formHasError = false;
+    $('#retrieval-timout').parent().addClass('has-error');
+  }
+  if (isNaN($('#max-flight-time').val())) { // Validated max-flight time
```
Comment (lwaghorn) [inline OnLand/ControlLaptop/static/assets/js/FlightProfile.js]: all times > 0

File: OnLand/ControlLaptop/static/assets/js/FlightProfile.js [hunk 4]
```diff
@@ -33,10 +74,25 @@ function setupConfigurationFormSubmit() {
       error: function handleErrorFormSubmission(data) {
         console.log("FAILED");
```
Comment (lwaghorn) [inline OnLand/ControlLaptop/static/assets/js/FlightProfile.js]: More descriptive error

File: OnLand/ControlLaptop/templates/_cards/profile.html [hunk 5]
```diff
@@ -90,7 +110,7 @@ <h4 class="title">Flight Profile</h4>
                <td>Brake Set A - Max PSI</td>
                <td id="brakeAMaxValue" >500</td>
                <td>PSI</td>
-               <td>Maximium Allowable Pressure in Brake set A</td>
+               <td>Maximum Allowable Pressure in Brake set A</td>
             </tr>
             <tr>
                <td><i id="brakeBStatus" class="fa fa-circle text-success"></i></td>
```
Comment (lwaghorn) [inline OnLand/ControlLaptop/templates/_cards/profile.html:116]: We should make the second table on this page its own card. Then we can easily load it on the dashboard page as "last sent config"

File: OnLand/ControlLaptop/static/assets/js/FlightProfile.js (hunk 4 above)
Comment (lwaghorn) [reply OnLand/ControlLaptop/static/assets/js/FlightProfile.js]: If the error from the server is connection refused then we should relay that. Will be very important to know that we aren't connecting to the pod

File: OnLand/ControlLaptop/views.py [hunk 6]
```diff
@@ -41,11 +42,22 @@ def inject_now():
...
+                'motor_speed': int(configuration_form.motor_speed.data),
+                'telemetry_port': int(configuration_form.telemetry_port.data),
+                'command_port': int(configuration_form.command_port.data),
+                'flight_length': int(configuration_form.flight_Length.data),
+                'heartbeat_timeout': int(configuration_form.heartbeat_timout.data),
+                'pod_address': configuration_form.pod_ip.data,
+                'pod_driver': 'Motor' if configuration_form.pod_driver.data is True else 'Simulation',
+            }
+        )
+        PodCommunicator.get_pod_communicator().send_configuration(
```
Comment (lwaghorn) [inline OnLand/ControlLaptop/views.py]: Right now if we try and send a config and theres no connection it just throws an error on the flask side and nothing gets relayed to the front end. sendConfiguration should be able to tell us: success, failed to connect, BB failed to prase. So we can then have descriptive failures on the front end. 

Maybe we create the PodCommunicator object when the app starts but we take the _connect_to_pod() method out of the constructor and make it its own call. The Flask App will likely be on well before the pod is on. When we load the profile page we try and connect and relay that info with the page. Then we have a button on the profile page that we can click and it simply runs the podCommunicator.connect() function and returns if it successfully connected. Or if we submit a profile it obvs tries to connect before sending.

It'd be a v simple add on and we can get a better idea of if the submission is going to be successful before we do submit.

File: OnLand/ControlLaptop/controlLaptop.py [hunk 7]
```diff
@@ -1,6 +1,5 @@
-#!/home/lwaghorn/Development/comp4-software/OnLand/venv/bin/python3
```
Comment (lwaghorn) [inline OnLand/ControlLaptop/controlLaptop.py:1]: sorry I should have taken that out before

**Review by lwaghorn — CHANGES_REQUESTED** [review]

File: OnLand/ControlLaptop/templates/_cards/profile.html (hunk 5 above)
Comment (mmb186) [reply OnLand/ControlLaptop/templates/_cards/profile.html:116]: Can be a task for Future Prs.

**Review by lwaghorn — APPROVED** [review]

**Pull request merged**